config/v1/types_build.go

```diff
@@ -32,54 +32,77 @@ type BuildSpec struct {
 }
 
 type BuildDefaults struct {
+	// RegistryProxy is the proxy settings for image pull/push
+	// +optional
+	RegistryProxy Proxy `json:"registryProxy,omitempty"`
 	// GitHTTPProxy is the location of the HTTPProxy for Git source
 	// +optional
 	GitHTTPProxy string `json:"gitHTTPProxy,omitempty"`
-
 	// GitHTTPSProxy is the location of the HTTPSProxy for Git source
 	// +optional
 	GitHTTPSProxy string `json:"gitHTTPSProxy,omitempty"`
-
-	// GitNoProxy is the list of domains for which the proxy should not be used
+	// GitNoProxy is the list of domains for which the proxy should not be used for Git source
 	// +optional
 	GitNoProxy string `json:"gitNoProxy,omitempty"`
-
 	// Env is a set of default environment variables that will be applied to the
 	// build if the specified variables do not exist on the build
 	// +optional
 	Env []corev1.EnvVar `json:"env,omitempty"`
-
 	// ImageLabels is a list of docker labels that are applied to the resulting image.
 	// User can override a default label by providing a label with the same name in their
 	// Build/BuildConfig.
 	// +optional
 	ImageLabels []ImageLabel `json:"imageLabels,omitempty"`
-
 	// Resources defines resource requirements to execute the build.
 	// +optional
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+	// RegistriesConfig controls the registries allowed for image pull and push.
+	// +optional
+	RegistriesConfig RegistriesConfig
 }
 
 type ImageLabel struct {
 	// Name defines the name of the label. It must have non-zero length.
 	Name string `json:"name"`
-
 	// Value defines the literal value of the label.
 	// +optional
 	Value string `json:"value,omitempty"`
 }
 
+type Proxy struct {
+	// HTTPProxy is the URL of this proxy for HTTP traffic
+	// +optional
+	HTTPProxy string `json:"httpProxy,omitempty"`
+	// HTTPSProxy is the URL of this proxy for HTTPS traffic
+	// +optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+	// NoProxy is the list of the domains for which this proxy should not be used
+	// +optional
+	NoProxy string `json:"noProxy,omitempty"`
+}
+
+type RegistriesConfig struct {
+	// SearchRegistries lists the registries to search for images if an image repository is not specified in an image pull spec
+	// +optional
+	SearchRegistries []string `json:"searchRegistries,omitempty"`
```

Comment:
@bparees I feel giving `RegistriesConfig` first-class support will be more consumer-friendly and resilient to change in the long run.

Reply:
seems reasonable to me, but it means you're going to have to reconstruct the registries.conf file in the build controller

Reply:
correct, but that isn't hard and isolates tooling changes from cluster admins (YAML yesterday, TOML today, tomorrow who knows...)